models/helper_environment.go

```diff
@@ -12,13 +12,13 @@ import (
 
 // PushingEnvironment returns an os environment to allow hooks to work on push
 func PushingEnvironment(doer *User, repo *Repository) []string {
-	return FullPushingEnvironment(doer, doer, repo, 0)
+	return FullPushingEnvironment(doer, doer, repo.MustOwnerName(), repo.Name, repo.ID, 0)
```

Comment:
```suggestion
	return FullPushingEnvironment(doer, doer, repo, false, 0)
```

Reply:
This will not work. we need `repoName` which is different from `repo.Name`. When it's a wiki, `repo.Name` will return `my_repo`, but `repoName` should be `my_repo.wiki`.